### ISSUE TITLE
NullPointerException in compileAndRunTests()

### ISSUE THREAD

**Issue by dasKeks** [bendisposto/virtual-kata-lib#5]
The method _compiler.compileAndRunTests();_ throws a NullPointerException if a method is called with a wrong signature:

**Example:**
Code

``` java
public class Foo {
    public static void foo(int bar){}
}
```

Test

``` java
import org.junit.Test;

public class FooTest {
    @Test
    public void testFoo(){
        Foo.foo("bar");
    }
}
```

**Expected:**
A vk.core.api.CompileError with the following cause:

```
java.lang.Error: Unresolved compilation problem: 
    The method foo(int) in the type Foo is not applicable for the arguments (String)
```

**Actual:**
NullPointerException in the virtual-kata-lib's _compiler.compileAndRunTests()_ Method.

```
Caused by: java.lang.NullPointerException
    at vk.core.internal.InternalCompiler.compileAllUnits(InternalCompiler.java:184)
    at vk.core.internal.InternalCompiler.compileAndRunTests(InternalCompiler.java:60)
```

**Comment by KozukeKamizu** [comment]
The NullPointerException is throwed by the virtual kata lib.
I can't handle the error:incompatible types, because i can't get the CompilerResult until the hole compilation is done and the field "compilerCalled" in the class "IntermalCompiler" is set to "true".

                
![debug](https://cloud.githubusercontent.com/assets/7966926/16896122/132ebdce-4b8b-11e6-969c-be5100468be6.jpg)

**Comment by bendisposto** [comment]
Fixed in version 2.0
